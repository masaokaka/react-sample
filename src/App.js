import React,{ useEffect,useState } from 'react';
import { useSelector,useDispatch } from 'react-redux';
import { fetchData , addToTodoList, doneTodo, deleteTodo } from './actions';
import { IconButton,Container,TextField,Checkbox,Grid} from '@material-ui/core'
import { Table,TableBody,TableCell,TableContainer,TableHead,TableRow,Paper} from '@material-ui/core'
import { AddCircle,Delete } from '@material-ui/icons';


//useSlector
const todosSelector = state => state.todo.todos;

const App = () => {
  const dispatch = useDispatch()
  const todos = useSelector(todosSelector)
  const [val,setVal] = useState('')

  //マウント時に実行
  useEffect(() => {
    if (localStorage.data) {
      let data = localStorage.getItem('data')
      data = JSON.parse(data)
      dispatch(fetchData(data))
    }
  }, [])

  //todosに変化があったときのみinputタグ内のテキストを削除
  useEffect(() => {
    setVal('')
  }, [todos])
  
  return (
    <Container maxWidth="Md">
      <h2>ToDoリスト</h2>
      {todos.length!==0 &&
      <TableContainer component={Paper}>
        <Table>
        <TableHead>
          <TableRow>
            <TableCell align="center">ID</TableCell>
            <TableCell align="center">テキスト</TableCell>
            <TableCell align="center">機能</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
        {todos.map((todo, index) =>
          <TableRow key={index}>
            <TableCell align="center">{todo.id}</TableCell>
            {todo.flg ? 
            <TableCell align="center"><del>{todo.value}</del></TableCell>
            :
            <TableCell align="center"><span>{todo.value}</span></TableCell>
            }
            <TableCell align="center">
              <Checkbox color="primary" type="checkbox" onChange={(() => { dispatch(doneTodo(index)) })} checked={todo.flg}/>
              <IconButton variant="contained" onClick={(() => { dispatch(deleteTodo(index)) })}>
                <Delete/>
              </IconButton>
            </TableCell>
          </TableRow>
        )}
        </TableBody>
        </Table>
        </TableContainer>
        }
        <Grid container justify="center" alignItems="center">
          <Grid item md={3}>
            <TextField variant="outlined" label="todoを入れてください" type="text" onChange={((e)=>{setVal(e.target.value)})} value={ val }/>
          </Grid>
          <Grid item md={1}>
            <AddCircle style={{fontSize:30}} color="primary" onClick={(() => { dispatch(addToTodoList(val)) })}></AddCircle>
          </Grid>
        </Grid>
    </Container>
  );  
}

export default App;