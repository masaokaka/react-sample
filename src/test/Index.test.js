import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers";

it("Page snapshot testing", () => {
    const store = createStore(reducer);
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
