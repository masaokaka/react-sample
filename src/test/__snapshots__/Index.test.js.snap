// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot testing 1`] = `
Array [
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          Todo List
        </h6>
      </div>
    </header>
  </div>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <h2>
      ToDoリスト
    </h2>
    <a
      href="/addtodo"
      onClick={[Function]}
    >
      追加
    </a>
  </div>,
  <div
    className="makeStyles-footer-4"
  >
    Footer
  </div>,
]
`;
